src/compiler/declarationEmitter.ts

```diff
@@ -1429,45 +1429,40 @@ namespace ts {
             function getAccessorDeclarationTypeVisibilityError(symbolAccessibilityResult: SymbolAccessibilityResult): SymbolAccessibilityDiagnostic {
                 let diagnosticMessage: DiagnosticMessage;
                 if (accessorWithTypeAnnotation.kind === SyntaxKind.SetAccessor) {
-                    // Setters have to have type named and cannot infer it so, the type should always be named
-                    if (hasModifier(accessorWithTypeAnnotation.parent, ModifierFlags.Static)) {

```

Comment:
Not sure what was up with the `.parent`, it didn't output the static error message when appropriate: https://github.com/Microsoft/TypeScript/pull/18593/files#diff-a085a7cb8abe9e3fd72630f034bbe686L165